drivers/power/lt3074/iio_lt3074.c

```diff
@@ -0,0 +1,727 @@
+*    this list of conditions and the following disclaimer.
+*
+* 2. Redistributions in binary form must reproduce the above copyright notice,
+*    this list of conditions and the following disclaimer in the documentation
+*    and/or other materials provided with the distribution.
+*
+* 3. Neither the name of Analog Devices, Inc. nor the names of its
+*    contributors may be used to endorse or promote products derived from this
+*    software without specific prior written permission.
+*
+* THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES, INC. “AS IS” AND ANY EXPRESS OR
+* IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+* MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO
+* EVENT SHALL ANALOG DEVICES, INC. BE LIABLE FOR ANY DIRECT, INDIRECT,
+* INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+* LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA,
+* OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+* LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+* NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+* EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include "no_os_alloc.h"
+#include "no_os_error.h"
+#include "no_os_units.h"
+#include "no_os_util.h"
+
+#include "lt3074.h"
+#include "iio_lt3074.h"
+
+static const int32_t lt3074_margin_avail[] = {
+	0,
+	1,
+	3,
+	5,
+	10,
+	15,
+	20,
+	25,
+	30,
+};
+
+enum lt3074_iio_chan_type {
+	LT3074_IIO_VIN_CHAN,
+	LT3074_IIO_VOUT_CHAN,
+	LT3074_IIO_IOUT_CHAN,
+	LT3074_IIO_TEMP_CHAN,
+	LT3074_IIO_VBIAS_CHAN,
+};
+
+static struct iio_device lt3074_iio_dev;
+
+/**
+ * @brief Read register value.
+ * @param dev     - The iio device structure.
+ * @param reg	  - Register to read.
+ * @param readval - Read value.
+ * @return ret    - Result of the reading procedure.
+ * 		    In case of success, the size of the read data is returned.
+*/
+static int32_t lt3074_iio_reg_read(void *dev, uint32_t reg, uint32_t *readval)
+{
+	struct lt3074_iio_desc *iio_lt3074 = dev;
+	struct lt3074_dev *lt3074 = iio_lt3074->lt3074_dev;
+	int ret;
+	uint16_t word;
+	uint8_t byte;
+	uint8_t block[6] = {0};
+
+	switch (reg) {
+	case LT3074_PAGE:
+	case LT3074_OPERATION:
+	case LT3074_ON_OFF_CONFIG:
+	case LT3074_WRITE_PROTECT:
+	case LT3074_STATUS_BYTE:
+	case LT3074_STATUS_VOUT:
+	case LT3074_STATUS_IOUT:
+	case LT3074_STATUS_INPUT:
+	case LT3074_STATUS_TEMPERATURE:
+	case LT3074_STATUS_CML:
+	case LT3074_STATUS_MFR_SPECIFIC:
+	case LT3074_MFR_MARGIN:
+	case LT3074_MFR_RAIL_ADDRESS:
+	case LT3074_CAPABILITY:
+	case LT3074_VOUT_MODE:
+	case LT3074_IOUT_OC_FAULT_RESPONSE:
+	case LT3074_REVISION:
+	case LT3074_MFR_DEFAULT_CONFIG:
+		ret = lt3074_read_byte(lt3074, (uint8_t)reg, &byte);
```

Comment:
this will definitely work but i expect this kind of register level access size to be handled in the device driver so at the lower level, not here

Reply:
this is how we prefer to do it:
https://github.com/analogdevicesinc/no-OS/blob/2d0e5d5ed9244bf91e4b47870702a696406aab71/drivers/adc/ad9250/ad9250.h#L54

each register has size information coded into the register "address" macro so the lowest level spi/i2c/whatever read/write, knows how many bytes to read/write